url64d634b8f0_agdturner_generic_tgz-pJ8-uk_ac_leeds_ccg_andyt_generic_visualisation_charts_Generic_BarChartJ8/src/uk/ac/leeds/ccg/andyt/generic/execution/AgeConverter.java

```diff
@@ -0,0 +1,585 @@
+            return 18;
+        }
+        if (age < 22) {
+            return 20;
+        }
+        if (age < 24) {
+            return 22;
+        }
+        if (age < 26) {
+            return 24;
+        }
+        if (age < 28) {
+            return 26;
+        }
+        if (age < 30) {
+            return 28;
+        }
+        if (age < 32) {
+            return 30;
+        }
+        if (age < 34) {
+            return 32;
+        }
+        if (age < 36) {
+            return 34;
+        }
+        if (age < 38) {
+            return 36;
+        }
+        if (age < 40) {
+            return 38;
+        }
+        if (age < 42) {
+            return 40;
+        }
+        if (age < 44) {
+            return 42;
+        }
+        if (age < 46) {
+            return 44;
+        }
+        if (age < 48) {
+            return 46;
+        }
+        if (age < 50) {
+            return 48;
+        }
+        if (age < 52) {
+            return 50;
+        }
+        if (age < 54) {
+            return 52;
+        }
+        if (age < 56) {
+            return 54;
+        }
+        if (age < 58) {
+            return 56;
+        }
+        if (age < 60) {
+            return 58;
+        }
+        if (age < 62) {
+            return 60;
+        }
+        if (age < 64) {
+            return 62;
+        }
+        if (age < 66) {
+            return 64;
+        }
+        if (age < 68) {
+            return 66;
+        }
+        if (age < 70) {
+            return 68;
+        }
+        if (age < 72) {
+            return 70;
+        }
+        if (age < 74) {
+            return 72;
+        }
+        if (age < 76) {
+            return 74;
+        }
+        if (age < 78) {
+            return 76;
+        }
+        if (age < 80) {
+            return 78;
+        }
+        return 80;
+    }
+
+    /**
+     * @param age
+     * @return short of age which is by default 97 and is otherwise:
+     * 0-15,16,20,25,30,45,60,65,75,76-96
+     */
+    public static short getAgeClassISARDataRecord(int age) {
+        if (age < 16) {
+            return (short) age;
+        }
+        if (age < 20) {
+            return 16;
+        }
+        if (age < 25) {
+            return 20;
+        }
+        if (age < 30) {
+            return 25;
+        }
+        if (age < 45) {
+            return 30;
+        }
+        if (age < 60) {
+            return 45;
+        }
+        if (age < 65) {
+            return 60;
+        }
+        if (age < 70) {
+            return 65;
+        }
+        if (age < 75) {
+            return 70;
+        }
+        if (age < 96) {
+            return (short) age;
+        }
+        return 97;
+    }
+
+    /**
+     * @param age
+     * @param aRandom
+     * @return int age which is by default 90 else a random assignment in ranges:
+     * 0-19,20-24,25-29,30-39,40-44,45-49,50-54,55-59,60-64,65-74,75-84,85-90.
+     */
+    public static int getAge1(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.